### PR TITLE
feat(payment): PI-1583 Write integration tests to verify all humm components of Checkout SDK/JS are working together

### PR DESCRIPTION
**Pull request by bc-sebastianszafraniec**
## What?
Added E2E tests for Humm integration on Checkout JS

## Testing / Proof
All tests have passed

@bigcommerce/team-checkout

### CODE REVIEW

File: packages/test-framework/src/fixture/pagePreset/ApiRequestsSender.ts [hunk 1]
```diff
@@ -15,13 +15,13 @@ export class ApiRequestsSender {
     private readonly storeUrl: string;
     private readonly startTime: number;
 
-    constructor(page: Page) {
+    constructor(page: Page, countryCode = 'US') {
```
[the remark this answers is not in the document]
Comment (animesh1987) [reply packages/test-framework/src/fixture/pagePreset/ApiRequestsSender.ts:18]: Or we can use the value from the enum in types file.

[the remark this answers is not in the document]
Comment (bc-peng) [reply packages/test-framework/src/fixture/pagePreset/ApiRequestsSender.ts:18]: API request sender does not need countryCode to work.
Does it make sense to name `countryCode` here as `fakerLocale`?

File: packages/hosted-payment-integration/e2e/support/humm.ejs
Comment (richapatel1510) [inline packages/hosted-payment-integration/e2e/support/humm.ejs:1]: Is this file getting used somewhere? I don't see it getting used in `packages/hosted-payment-integration/e2e/humm.spec.ts`

File: packages/test-framework/src/fixture/pagePreset/ApiRequestsSender.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bc-sebastianszafraniec) [reply packages/test-framework/src/fixture/pagePreset/ApiRequestsSender.ts:18]: @bc-peng I will rename `conutryCode` to `fakerLocale`

File: packages/test-framework/src/fixture/pagePreset/PaymentStepAsGuestAU.ts
[the remark this answers is not in the document]
Comment (bc-sebastianszafraniec) [reply packages/test-framework/src/fixture/pagePreset/PaymentStepAsGuestAU.ts:1]: I tried but it didn't worked. The problem was that during test the state/province field was not filled automaticly and for Australia, this field is required. I needed to use `'en_AU'` Locale in order to make it work and thats why I created `PaymentStepAsGuestAU` preset.

File: packages/hosted-payment-integration/e2e/support/humm.ejs
Comment (bc-sebastianszafraniec) [reply packages/hosted-payment-integration/e2e/support/humm.ejs:1]: @richapatel1510 you are right. I deleted that file as it is not needed.

File: packages/test-framework/src/fixture/pagePreset/PaymentStepAsGuestAU.ts
[the remark this answers is not in the document]
Comment (bc-peng) [reply packages/test-framework/src/fixture/pagePreset/PaymentStepAsGuestAU.ts:1]: I see. Is it possible to meet your needs by changing this line?
https://github.com/bigcommerce/checkout-js/blob/b1ba3d8ec9d41dfaca28206f535952b60d052209/packages/test-framework/src/fixture/pagePreset/PaymentStepAsGuest.ts#L9

[the remark this answers is not in the document]
Comment (bc-sebastianszafraniec) [reply packages/test-framework/src/fixture/pagePreset/PaymentStepAsGuestAU.ts:1]: Yes, I pushed the changes with such a solution

**Review by richapatel1510 — APPROVED** [review]
LGTM 👏

**Review by bc-peng — APPROVED** [review]
LGTM.

**Pull request merged**